src/tests/ftest/launch.py

```diff
@@ -20,164 +20,550 @@
+
+    This host will be included in the list if no clients are explicitly called
+    out in the test's yaml file.
+
+    Args:
+        test_yaml (str): test yaml file
+
+    Returns:
+        list: a unique list of hosts specified in the test's yaml file
+
+    """
+    yaml_data = get_yaml_data(test_yaml)
+    server_keys = [
+        "test_machines",
+        "test_servers",
+        "daos_servers",
+        "servers",
+        "test_machines1",
+        "test_machines2",
+        "test_machines2a",
+        "test_machines3",
+        "test_machines6"]
+    client_keys = [
+        "test_clients",
+        "clients",
+    ]
+    host_set = set()
+    found_client_key = False
+    matches = find_values(yaml_data, server_keys + client_keys)
+    for key, value in matches.items():
+        host_set.update(value)
+        if key in client_keys:
+            found_client_key = True
+
+    # Include this host as a client if no clients are specified
+    if not found_client_key:
+        host_set.add(socket.gethostname().split(".")[0])
+
+    return sorted(list(host_set))
+
+
+def clean_logs(test_yaml):
+    """Remove the test log files on each test host.
+
+    Args:
+        test_yaml (str): yaml file containing host names
+    """
+    # Use the default server yaml and then the test yaml to update the default
+    # DAOS log file locations.  This should simulate how the test defines which
+    # log files it will use when it is run.
+    log_files = get_log_files(test_yaml, get_log_files(BASE_LOG_FILE_YAML))
+    host_list = get_hosts_from_yaml(test_yaml)
+    command = "ssh {{host}} \"rm -fr {}\"".format(" ".join(log_files.values()))
+    print("Cleaning logs on {}".format(host_list))
+    spawn_commands(host_list, command)
+
+
+def archive_logs(avocado_logs_dir, test_yaml):
+    """Copy all of the host test log files to the avocado results directory.
+
+    Args:
+        avocado_logs_dir ([type]): [description]
+        test_yaml (str): yaml file containing host names
+    """
+    this_host = socket.gethostname().split(".")[0]
+    log_files = get_log_files(TEST_LOG_FILE_YAML)
+    host_list = get_hosts_from_yaml(test_yaml)
+    doas_logs_dir = os.path.join(avocado_logs_dir, "latest", "daos_logs")
+
+    # Create a subdirectory in the avocado logs directory for this test
+    print("Archiving host logs from {} in {}".format(host_list, doas_logs_dir))
+    get_output("mkdir {}".format(doas_logs_dir))
+
+    # Create a list of log files that are not directories
+    non_dir_files = [
+        log_file for log_file in log_files.values()
+        if os.path.splitext(os.path.basename(log_file))[1] != ""]
+
+    # Copy any log files that exist on the test hosts and remove them from the
+    # test host if the copy is successful
+    command = "set -e; ssh {{host}} '{}'".format(
+        "for file in {}; do if [ -e $file ]; then "
+        "scp $file {}:{}/{{host}}-${{{{file##*/}}}} && rm -fr $file; fi; "
```

Comment:
I didn't see it until I looked at [this artifact dir](https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-596/4/artifact/Functional_Hardware/job-results/job-2019-06-13T03.02-4603c24-daos_core_test/daos_logs/) but I wonder if the logs would be better organised if the hostname was at the end instead of the beginning.  I think generally speaking people want to see the same kinds of logs together rather than the logs from the same host together.

Reply:
I moved appending the host name to the end of `.log` extension, i.e. `daos_agent.log-boro-41`

Reply:
I think it will be more obvious how useful that is when we can see the Functional/hw daos_core_test results, but even in VM test results, I think it's better.